packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -425,7 +426,7 @@ export class AngularWebpackPlugin {
       for (const sourceFile of builder.getSourceFiles()) {
         // Collect Angular template diagnostics
         if (!ignoreForDiagnostics.has(sourceFile)) {
-          diagnosticsReporter(angularCompiler.getDiagnostics(sourceFile));
+          diagnosticsReporter(angularCompiler.getDiagnosticsForFile(sourceFile, OptimizeFor.WholeProgram));
```

Comment:
Do you need a fallback to `getDiagnostics()` if the new API is not available?

Reply:
The Ivy plugin is only available in CLI version 11.1. I believe this doesn't work properly Angular 11.0. But I might be wrong.

@clydin would know more.

Reply:
It’s unlikely but someone could use 11.0 of the compiler with 11.1 of the CLI.  So a guard and some creative casting should ideally be present here.

Reply:
Updated